Containerfile.tests

```diff
@@ -0,0 +1,26 @@
+#
+# Copyright (c) 2023 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+#
+
+FROM docker.io/golang:1.21 AS builder
+WORKDIR /go/src/github.com/openshift-kni/oran-o2ims
+COPY . .
+
+# Install ginkgo & mockgen.
+RUN go install github.com/onsi/ginkgo/v2/ginkgo@$(go list -f '{{.Version}}' -m github.com/onsi/ginkgo/v2) && \
+  go install go.uber.org/mock/mockgen@v0.3.0
+
+# Install golangci-lint.
+RUN curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.55.2 && \
+  golangci-lint --version
```

Comment:
This way to install golangci-lint is a bit scary: it runs a shell script downloaded from GitHub that we don't really know what will do. Can we instead download the release artifact, verify that it hasn't changed since we checked the content and install it? For example:

```
RUN \
  curl -Lo tarball https://github.com/golangci/golangci-lint/releases/download/v1.55.2/golangci-lint-1.55.2-linux-amd64.tar.gz && \
  echo ca21c961a33be3bc15e4292dc40c98c8dcc5463a7b6768a3afc123761630c09c tarball | sha256sum -c && \
  tar -C /usr/local/bin --strip-components=1 -xf tarball golangci-lint-1.55.2-linux-amd64/golangci-lint && \
  rm tarball
```

The point there is that you are supposed to manually download the file, check that it is correct and then recalculate the sha256sum yourself. That way, if the file were to change (intentionally or by accident) the build will fail.